src/Runner.Worker/StepsRunner.cs

```diff
@@ -120,6 +120,11 @@ public async Task RunAsync(IExecutionContext jobContext)
                     {
                         // Set GITHUB_ACTION
                         step.ExecutionContext.SetGitHubContext("action", actionStep.Action.Name);
+                        // Set GITHUB_ACTION_REPOSITORY if this Action is from a repository
+                        if (actionStep.Action.Reference is Pipelines.RepositoryPathReference repositoryReference)
+                        {
+                            step.ExecutionContext.SetGitHubContext("action_repository", repositoryReference.Name);
```

Comment:
I notice the `RepositoryPathReference` also has a `Ref` field. Would it be ok to also make that available, perhaps as a `action_ref` var? That would be very useful for code scanning as we've been wanting to include that data in our status reports, but couldn't work out how to get it. Doesn't have to be in this PR if the extra change would slow it down.

Reply:
Yes, sure. I've added that field as well.